raymon/tests/profiling/test_element.py

```diff
@@ -0,0 +1,45 @@
+from raymon.profiling.extractors.structured import ElementExtractor, generate_components
+from raymon.profiling.components import InputComponent
+import pandas as pd
+import numpy as np
+
+
+def test_element():
+    extractor = ElementExtractor(3)
+    extractor.element = 4
+    assert extractor.element == 4
+
+
+def test_extract():
+    vector1 = np.array([[10], [20], [30]])
+    assert ElementExtractor(2).extract(vector1) == [30]
+    cols = {
+        "num1": list(range(10)),
+        "cat1": ["a"] * 5 + ["b"] * 5,
+        "cat2": ["c"] * 5 + ["d"] * 5,
+        "num2": [0.2] * 10,
+    }
+    assert ElementExtractor("num1").extract(cols) == list(range(10))
```

Comment:
This test kind of makes sense, but you should really pass one row of the DF to the extractor. Something like `ElementExtractor("num1").extract(cols.iloc[0, :])`